File: internal/sast/fixtures/javascript/xxe_negative.js
// internal/sast/fixtures/javascript/xxe_negative.js
const libxmljs = require("libxmljs");
const express = require("express");
const app = express();
app.use(express.text({ type: "application/xml" }));

app.post("/parse", (req, res) => {
  const doc = libxmljs.parseXmlString(req.body);
  res.json({ root: doc.root().name() });
});

File: internal/sast/fixtures/csharp/Xxe_negative.cs
// internal/sast/fixtures/csharp/Xxe_negative.cs
using System.Xml;
using Microsoft.AspNetCore.Mvc;

public class XxeNegativeController : Controller
{
    public IActionResult Parse(string xml)
    {
        var doc = new XmlDocument { XmlResolver = null };
        doc.LoadXml(xml);
        return Content(doc.DocumentElement.Name);
    }
}
